File: bug39_no_base_case/solved/main.java
public class main {
    public static int infinite_recursion(int n) { // correct part
        if (n <= 0) return 0; // correct part
        return n + infinite_recursion(n - 1);
    }
    public static void main(String[] args) {
        System.out.println(infinite_recursion(10));
    }
}

File: bug39_no_base_case/solved/main.cpp
#include <iostream>

int infinite_recursion(int n) {
    if (n <= 0) return 0; // correct part
    return n + infinite_recursion(n - 1);
}

int main() {
    std::cout << infinite_recursion(10) << std::endl;
    return 0;
}
